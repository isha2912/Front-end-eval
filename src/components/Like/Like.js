import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import getData from '../../utils/axiosUtil';

const Like = (({ eachSong }) => {
  const [likeData, setLikeData] = useState({});

  useEffect(async () => {
    const like = await getData.getLikes(eachSong.id);
    setLikeData(like.data);
  }, []);
  const handleLike = (async () => {
    const isLiked = likeData.like;
    const newLike = {
      like: (!isLiked),
    };

    const updateLike = await getData.updateLike(eachSong.id, newLike);
    setLikeData(updateLike.data);
    console.log(likeData);
  });

  return (
    <button type="button" onClick={handleLike}>

      <img src={likeData.like === true ? './assets/heart-red.svg' : './assets/heart-gray.svg'} alt="heart" />
      {likeData.count}

    </button>
  );
}
);
export default Like;
Like.propTypes = {
  eachSong: PropTypes.shape({
    id: PropTypes.string.isRequired,

  }).isRequired,
};
