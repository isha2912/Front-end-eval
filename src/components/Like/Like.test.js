import { render } from '@testing-library/react';
import React from 'react';
import Like from './Like';

describe(Like.name, () => {
  const mockSong = {
    eachSong: {
      name: 'yariyan',
      artist: {
        name: 'Arijit singh',
      },
      albumArtUrl: 'url',
      id: '9',
    },
  };
  test('Should match screenshot', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const { container } = render(<Like {...mockSong} />);
    expect(container).toMatchSnapshot();
  });
});
