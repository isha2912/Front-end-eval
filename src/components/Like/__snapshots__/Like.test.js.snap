// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like Should match screenshot 1`] = `
<div>
  <button
    type="button"
  >
    <img
      alt="heart"
      src="./assets/heart-gray.svg"
    />
  </button>
</div>
`;
